docs(docs): nx cloud e2e encryption configuration documentation added (#3916)
nrwl/nx: docs/angular/guides/configuration.md

- `captureStderr` defines whether the cache will capture stderr or just stdout
- `skipNxCache` defines whether the Nx Cache should be skipped. Defaults to `false`
- `cacheDirectory` defines where the local cache is stored, which is `node_modules/.cache/nx` by default.

> - `encryptionKey` (when using `"@nrwl/nx-cloud"` only) defines an encryption key to support end-to-end encryption of your cloud cache. See also [End-to-end Cloud Encryption](#end-to-end-cloud-encryption) for providing this key via an environment variable. The Nx Cloud task runner will normalize the key length, so any length of key is acceptable.

Reviewer: ```suggestion
- `encryptionKey` (when using `"@nrwl/nx-cloud"` only) defines an encryption key to support end-to-end encryption of your cloud cache. You may also provide an environment variable with the key `NX_CLOUD_ENCRYPTION_KEY` that contains an encryption key as its value. The Nx Cloud task runner will normalize the key length, so any length of key is acceptable.
```

> - `encryptionKey` (when using `"@nrwl/nx-cloud"` only) defines an encryption key to support end-to-end encryption of your cloud cache. See also [End-to-end Cloud Encryption](#end-to-end-cloud-encryption) for providing this key via an environment variable. You may also provide an environment variable with the key `NX_CLOUD_ENCRYPTION_KEY` that contains an encryption key as its value. The Nx Cloud task runner will normalize the key length, so any length of key is acceptable.

Reviewer: ```suggestion
- `encryptionKey` (when using `"@nrwl/nx-cloud"` only) defines an encryption key to support end-to-end encryption of your cloud cache. You may also provide an environment variable with the key `NX_CLOUD_ENCRYPTION_KEY` that contains an encryption key as its value. The Nx Cloud task runner will normalize the key length, so any length of key is acceptable.
```

1. Changes to that file will not be taken into account in the `affected` calculations.
2. Even if the file is outside an app or library, `nx workspace-lint` will not warn about it.

> ## End-to-end Cloud Encryption

Reviewer: This duplicates a lot of the text in the section above.  I would remove this section and add a not to the command line option saying "You may also provide this value as an environment variable with the key: `NX_CLOUD_ENCRYPTION_KEY`"

Author: Moved - good call
